src/pages/studyView/table/FixedHeaderTable.tsx

```diff
@@ -351,37 +351,48 @@ export default class FixedHeaderTable<T> extends React.Component<
     }
 
     getAddRemoveAllButton() {
+        const noneSelected = this.props.numberOfSelectedRows === 0;
         const allSelected =
             this.props.numberOfSelectedRows === this.props.data.length;
 
-        let dataTest: string, onClick: () => void, content: string;
-        let showButton = false;
-
-        if (allSelected && this.props.removeAll) {
-            dataTest = 'fixed-header-table-remove-all';
-            onClick = this.onRemoveAll;
-            content = 'Deselect all';
-            showButton = true;
-        } else if (this.props.addAll) {
-            dataTest = 'fixed-header-table-add-all';
-            onClick = this.onAddAll;
-            content = `Select all${
+        let selectAllContent: string;
+        let showDeselectAll = false;
+        let showSelectAll = false;
+
+        if (!noneSelected && this.props.removeAll) {
+            showDeselectAll = true;
+        }
+        if (!allSelected && this.props.addAll) {
+            showSelectAll = true;
+            selectAllContent = `Select all${
                 this.props.showSelectableNumber
                     ? ` (${this._store.dataStore.sortedFilteredData.length})`
                     : ''
```

Comment:
I appreciate the sentiment of only calculating this string when needed, but I don't think it's necessary. The performance savings is negligible, and I'd rather not spend time thinking when that string is or isn't computed. Let's move this logic to the declaration of `selectAllContent` and make that variable a const